### PR TITLE
Notification

### PR DESCRIPTION
**Pull request by namanahuja**

### CODE REVIEW

File: src/controllers/services/getTemplateParamsController.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+var notificationTemplateService = require('../../services/notificationTemplateService')
```
Comment (sahil-proptiger) [inline src/controllers/services/getTemplateParamsController.js]: no need to to create a new service for this. add this in notificationTemplateService.

File: src/services/notificationTemplateService.js [hunk 2]
```diff
@@ -100,3 +92,16 @@ module.exports.getTemplate = function(id){
     });
 }
 
+
+module.exports.getTemplateParams = function(type,medium){
```
Comment (sahil-proptiger) [inline src/services/notificationTemplateService.js]: this function should be getTemplateByMediumAndType

File: app.js [hunk 3]
```diff
@@ -18,9 +18,64 @@ app.get('/add', require('./src/controllers/pages/addNotificationController'))
...
+
 
 
 
 app.listen(5000, function() {
     console.log("Started server at port 5000")
 });
+
+
+request({
```
Comment (sahil-proptiger) [inline app.js]: Add this logic to a new service i.e. apiService

Comment (namanahuja) [inline app.js]: I was just testing the API calls from request module and AJAX (in home template). I will remove them once I start building the send module.

File: public/styles/bootstrap.min.css [hunk 4]
```diff
@@ -5024,7 +5024,7 @@ textarea.input-group-sm>.input-group-btn>.btn {
 .navbar {
     position: relative;
     min-height: 50px;
-    margin-bottom: 20px;
+
```
Comment (sahil-proptiger) [inline public/styles/bootstrap.min.css]: Don't make changes in this file.

File: app.js [hunk 5]
```diff
@@ -15,12 +15,69 @@ app.use(express.static('public'));
...
 app.get('/add', require('./src/controllers/pages/addNotificationController'))
-app.post('/addtodb', require('./src/controllers/services/insertNewController'))
-app.get('/edit/:id', require('./src/controllers/pages/editNotificationController'));
-app.post('/updatedb', require('./src/controllers/services/updateNotificationController'))
+app.post('/addtodb', require('./src/controllers/apis/insertNewController'))
+app.post('/updatedb', require('./src/controllers/apis/updateNotificationController'))
+app.get('/send', require('./src/controllers/pages/sendNotificationController'))
+app.post('/getparams', require('./src/controllers/apis/getTemplateParamsController'))
+app.post('/getTemplate', require('./src/controllers/apis/getTemplateController'))
+app.post('/deleteTemplate', require('./src/controllers/apis/deleteTemplateController'))
```
Comment (sahil-proptiger) [inline app.js]: Include Id in the params

File: app.js [hunk 6]
```diff
@@ -13,14 +13,28 @@ app.use(bodyParser.urlencoded({ limit: '50mb',extended: false }));
...
 app.get('/', require('./src/controllers/pages/notificationTemplateController'))
 app.get('/add', require('./src/controllers/pages/addNotificationController'))
-app.post('/addtodb', require('./src/controllers/services/insertNewController'))
-app.get('/edit/:id', require('./src/controllers/pages/editNotificationController'));
-app.post('/updatedb', require('./src/controllers/services/updateNotificationController'))
+app.get('/send', require('./src/controllers/pages/sendNotificationController'))
+
+//apis
+app.post('/emailTemplate', require('./src/controllers/apis/insertNewController'))
+app.put('/emailTemplate/:id', require('./src/controllers/apis/updateNotificationController'))
```
Comment (sahil-proptiger) [inline app.js]: name it updateTemplateController

File: app.js [hunk 7]
```diff
@@ -13,14 +13,28 @@ app.use(bodyParser.urlencoded({ limit: '50mb',extended: false }));
...
+//pages
 app.get('/', require('./src/controllers/pages/notificationTemplateController'))
 app.get('/add', require('./src/controllers/pages/addNotificationController'))
-app.post('/addtodb', require('./src/controllers/services/insertNewController'))
-app.get('/edit/:id', require('./src/controllers/pages/editNotificationController'));
-app.post('/updatedb', require('./src/controllers/services/updateNotificationController'))
+app.get('/send', require('./src/controllers/pages/sendNotificationController'))
+
+//apis
+app.post('/emailTemplate', require('./src/controllers/apis/insertNewController'))
```
Comment (sahil-proptiger) [inline app.js]: name it insertTemplateController

File: src/controllers/apis/getTemplateParamsController.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
+var notificationTemplateService = require('../../services/notificationTemplateService')
```
Comment (sahil-proptiger) [inline src/controllers/apis/getTemplateParamsController.js]: I don't think you need this file.

File: app.js [hunk 9]
```diff
@@ -13,14 +13,28 @@ app.use(bodyParser.urlencoded({ limit: '50mb',extended: false }));
 
 app.use(express.static('public'));
 
+//pages
 app.get('/', require('./src/controllers/pages/notificationTemplateController'))
```
Comment (sahil-proptiger) [inline app.js]: name is at viewTemplateController

File: app.js [hunk 10]
```diff
@@ -13,14 +13,28 @@ app.use(bodyParser.urlencoded({ limit: '50mb',extended: false }));
...
 app.get('/add', require('./src/controllers/pages/addNotificationController'))
```
Comment (sahil-proptiger) [inline app.js]: name is at addTemplateController

**Pull request merged**